src/silicone/database_crunchers/equal_quantile_walk.py

```diff
@@ -59,6 +75,16 @@ def derive_relationship(self, variable_follower, variable_leaders):
         data_follower_time_col = iamdf_follower.time_col
         data_follower_unit = iamdf_follower["unit"].values[0]
         lead_ts = self._db.filter(variable=variable_leaders).timeseries()
+        if not isinstance(weighting, type(None)):
```

Comment:
```suggestion
        if weighting is not None:
```

Reply:
This creates errors when weighting is a dictionary

Reply:
The tests don't fail if I make the change... https://github.com/GranthamImperial/silicone/pull/128/checks?check_run_id=2462978833

Reply:
indeed it doesn't, don't know how I triggered the error now